src/checkout-buttons/strategies/paypalCommerce/paypalCommerce-button-strategy.ts

```diff
@@ -0,0 +1,82 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { INTERNAL_USE_ONLY } from '../../../common/http-request';
+import { ApproveDataOptions, PaypalCommerceScriptLoader } from '../../../payment/strategies/paypalCommerce';
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _formPoster: FormPoster,
+        private _requestSender: RequestSender
+    ) {}
+
+    initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
```

Comment:
I'm happy for you to use `async/await` if you find it more readable. But if you're happy with this, I'm also good with it. 🍹

Reply:
Thank you!
Yes, it's more